Allow domain sockets to configure SO_SNDBUF and SO_RCVBUF

ellisjoe opened the pull request:
Motivation:

Running tests with a `KQueueDomainSocketChannel` showed worse performance than an `NioSocketChannel`. It turns out that the default send buffer size for Nio sockets is 64k while for KQueue sockets it's 8k. I verified that manually setting the socket's send buffer size improved perf to expected levels.

Modification:

Plumb the `SO_SNDBUF` and `SO_RCVBUF` options into the `DomainSocketChannelConfig`.

Result:

Can now configure send and receive buffer sizes for domain sockets.

normanmaurer commented on transport-native-unix-common/src/main/java/io/netty/channel/unix/DomainSocketChannelConfig.java:
```diff
@@ -77,4 +79,24 @@
      * Return the {@link DomainSocketReadMode} for the channel.
      */
     DomainSocketReadMode getReadMode();
```
We can’t make these changes in a point release :( That said we can keep the ones on the actual implementation. Please revert this part

ellisjoe replied:
removed. is there somewhere you'd like to track adding this in a future release?

normanmaurer commented on transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDomainSocketChannelConfig.java:
```diff
@@ -52,6 +53,12 @@
         if (option == ALLOW_HALF_CLOSURE) {
             return (T) Boolean.valueOf(isAllowHalfClosure());
         }
+        if (option == SO_SNDBUF) {
```
The same also needs to be done for setOption

normanmaurer commented on transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueDomainSocketChannelConfig.java:
```diff
@@ -52,6 +53,12 @@
             return (T) Boolean.valueOf(isAllowHalfClosure());
         }
+        if (option == SO_SNDBUF) {
+            return (T) Integer.valueOf(getSendBufferSize());
```
The same also needs to be done for setOption

normanmaurer commented on transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDomainSocketChannelConfig.java:
```diff
@@ -24,9 +24,10 @@
 import java.util.Map;
 
-import static io.netty.channel.ChannelOption.ALLOW_HALF_CLOSURE;
+import static io.netty.channel.ChannelOption.*;
```
please revert this... we usually not do wildcard imports for such things

ellisjoe commented:
thanks for the review @normanmaurer. I signed the icla and addressed your comments. Let me know if there's anything else you'd like me to follow up with.

normanmaurer commented:
I will just do it directly in master as a followup 

> Am 20.09.2019 um 15:49 schrieb Joe Ellis <notifications@github.com>:
> 
> ﻿
> @ellisjoe commented on this pull request.
> 
> In transport-native-unix-common/src/main/java/io/netty/channel/unix/DomainSocketChannelConfig.java:
> 
> > @@ -77,4 +79,24 @@
>       * Return the {@link DomainSocketReadMode} for the channel.
>       */
>      DomainSocketReadMode getReadMode();
> removed. is there somewhere you'd like to track adding this in a future release?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.

normanmaurer commented on transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDomainSocketChannelConfig.java:
```diff
@@ -173,4 +186,42 @@ public EpollDomainSocketChannelConfig setAllowHalfClosure(boolean allowHalfClosu
         return this;
     }
+
+    @Override
```
You need to remove the `@Override` annotations now

normanmaurer commented on transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueDomainSocketChannelConfig.java:
```diff
@@ -159,6 +173,44 @@ public DomainSocketReadMode getReadMode() {
         return mode;
     }
 
+    @Override
```
You need to remove the `@Override` annotations now

normanmaurer commented:
@netty-bot test this please

normanmaurer commented:
@ellisjoe thanks a lot!

normanmaurer commented:
@ellisjoe followup is here https://github.com/netty/netty/pull/9602
